app/Foliage/CmdBuild.hs

```diff
@@ -222,6 +230,42 @@ buildAction
             timestampInfoSnapshot = snapshotInfo
           }
 
+makeMetadataFile :: FilePath -> [PackageVersionMeta] -> Action ()
+makeMetadataFile outputDir packageVersions = traced "writing metadata" $ do
+  createDirectoryIfMissing True (outputDir </> "foliage")
+  Aeson.encodeFile
+    (outputDir </> "foliage" </> "packages.json")
+    ( Map.fromList
+        [ (prettyShow pkgId, encodeMetadataSpec pkgSpec)
+          | PackageVersionMeta pkgId pkgSpec <- packageVersions
+        ]
+    )
+
+encodeMetadataSpec :: PackageVersionSpec -> Aeson.Value
```

Comment:
Dumb question: why isn't this just the `ToJSON` instance for `PackageVersionSpec`?

Reply:
I wanted it to be different and stable so we don't accidentally break the schema. There are going to be tools relying on it.